### PR TITLE
AL - Re-add ESLint and fix linting errors

### LINKED ISSUE #2: Try to add eslint back into the workflow and into the CI/CD checks

### PR DESCRIPTION
**Pull request by andrewhlu**
This PR re-introduces ESLint, which was removed in ucsb-cs156-w22/demo-spring-react-example-v2#75 as a temporary fix to build problems on Heroku. The changes in that PR are partially reverted, and all linting issues are fixed.

I'm still not entirely sure why it broke in the first place, but everything seems to work now, so I'm going to assume all is well. Perhaps it's because of the Node version upgrades.

Note that this PR includes changes from #9, where the aforementioned Node upgrade / version consistency changes were implemented. Please merge that PR before merging this one.

Closes #2

### CODE REVIEW

File: frontend/src/main/components/Students/StudentsTable.js [hunk 1]
```diff
@@ -22,14 +21,12 @@ export default function StudentsTable({ students, currentUser }) {
         }
     ];
 
-    // Stryker disable ArrayDeclaration : [columns] and [students] are performance optimization; mutation preserves correctness
-    const memoizedColumns = React.useMemo(() => columns, [columns]);
-    const memoizedDates = React.useMemo(() => students, [students]);
-    // Stryker enable ArrayDeclaration
+    // Stryker disable next-line ArrayDeclaration : [students] is a performance optimization
+    const memoizedDates = useMemo(() => students, [students]);
```
Comment (btk5h) [inline frontend/src/main/components/Students/StudentsTable.js]: While you're cleaning up this code, you might as well remove this line as well. It's an improper usage of `useMemo` for the same reason ESLint complained about the other line.

File: frontend/src/main/components/UCSBDates/UCSBDatesTable.js [hunk 2]
```diff
@@ -46,18 +43,22 @@ export default function UCSBDatesTable({ dates, currentUser }) {
...
-    } 
+    const columnsIfAdmin = [
+        ...columns,
+        ButtonColumn("Edit", "primary", editCallback, "UCSBDatesTable"),
+        ButtonColumn("Delete", "danger", deleteCallback, "UCSBDatesTable")
+    ]
+
+    const columnsToDisplay = () => {
+        return hasRole(currentUser, "ROLE_ADMIN") ? columnsIfAdmin : columns;
+    }
```
Comment (btk5h) [inline frontend/src/main/components/UCSBDates/UCSBDatesTable.js]: Why are you wrapping this value in a function if you are just going to call it immediately? Can't you just replace this with 
```suggestion
    const columnsToDisplay = hasRole(currentUser, "ROLE_ADMIN") ? columnsIfAdmin : columns;
```

File: frontend/src/main/components/UCSBDates/UCSBDatesTable.js [hunk 3]
```diff
@@ -46,18 +43,22 @@ export default function UCSBDatesTable({ dates, currentUser }) {
...
 
-    // Stryker disable next-line ArrayDeclaration : [columns] is a performance optimization
-    const memoizedColumns = React.useMemo(() => columns, [columns]);
-    const memoizedDates = React.useMemo(() => dates, [dates]);
+    // Stryker disable next-line ArrayDeclaration : [dates] is a performance optimization
+    const memoizedDates = useMemo(() => dates, [dates]);
```
Comment (btk5h) [inline frontend/src/main/components/UCSBDates/UCSBDatesTable.js]: We should remove this line, it's another incorrect usage of `useMemo`

**Review by btk5h — CHANGES_REQUESTED** [review]

File: frontend/src/main/components/UCSBDates/UCSBDatesTable.js (hunk 2 above)
Comment (andrewhlu) [reply frontend/src/main/components/UCSBDates/UCSBDatesTable.js]: Good question. I implemented your change just now.

**Review by btk5h — APPROVED** [review]

**Pull request merged**